### DISCUSSION TITLE
How to change the default  open extension

### DISCUSSION THREAD

**Discussion by pingbaohua** [eclipse-theia/theia#14101] [Q&A]
After theia is running, how do I make the installed extensions open by default at startup, such as the Oracle Developer Tools for VS Code extension or other

**Comment by msujew** [comment]
Hey @pingbaohua,

please open such questions as discussions - we try to keep our issues for actionable items.

We dynamically register a command for each view contributed by VS Code extensions. See:

https://github.com/eclipse-theia/theia/blob/4c7ffb632b46f52841f09e6fdc88e9c23448b572/packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts#L424-L426

As far as I can tell the Oracle extension contributes those views:

![image](https://github.com/user-attachments/assets/8c17a7c8-8c0b-4447-90e1-ad9e7a0c459a)

You can tell call the command `oracleCloudExplorer.focus` on startup of your app.

**Reply by pingbaohua** [reply]
I didn't check Oracle debug because the Settings above didn't seem to work

**Reply by msujew** [reply]
I'm sorry, what?

**Reply by pingbaohua** [reply]
I tried setting this up but it didn't work： oracleCloudExplorer.focus，the extension selected last time is still displayed, such as Search extension

**Reply by msujew** [reply]
Can you share the code you used to do this?
